addons/hitreactions/functions/fnc_checkWeaponDrop.sqf

```diff
@@ -0,0 +1,32 @@
+#include "..\script_component.hpp"
+/*
+ * Author: KJW
+ * Checks if unit should drop weapon based on projectile hit info
+ *
+ * Arguments:
+ * 0: surfaceType <STRING>
+ * 1: components <ARRAY>
+ * 2: position <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["", [], [1,1,1]] call ACE_hitreactions_fnc_checkWeaponDrop
+ *
+ * Public: No
+ */
+
+params ["_surfaceType", "_components", "_position"];
+
+private _didHitArm = _components findAny GVAR(armComponents) isNotEqualTo -1;
+private _didHitGun = _surfaceType isEqualTo "" && _components isEqualTo [];  // Could be potential for edge cases here however couldn't find any from testing.
+
+if (!_didHitArm && !_didHitGun) exitWith {}; // Exit if it didn't hit an arm and also didn't hit a gun.
+
+private _unit = nearestObject [_position, "CAManBase"];
+private _weapon = currentWeapon _unit;
+
+if (_weapon in GVAR(undroppableGuns)) exitWith {};
```

Comment:
Where is this GVAR's default value set?

Reply:
I don't think so?

Standby. There's a safer way to define that var.

Reply:
Ref https://github.com/acemod/ACE3/blob/eee86a4baccf07626d035892e55132ea135fc05e/addons/medical_engine/XEH_preInit.sqf#L12

That way we don't overwrite user's modifications but still define the variable. Can do this in postInit.

```
if (isNil QGVAR(undroppableGuns)) then {GVAR(undroppableGuns = []};
```

Reply:
done